pkg/controller/direct/bigqueryanalyticshub/mapper.go

```diff
@@ -239,7 +239,41 @@ func BigQueryAnalyticsHubListingSpec_ToProto(mapCtx *direct.MapContext, in *krmv
 	dtype := direct.Enum_ToProto[pb.DiscoveryType](mapCtx, in.DiscoveryType)
 	out.DiscoveryType = &dtype
 
-	// todo acpana do source
+	out.Source = Listing_BigQueryDatasetSource_ToProto(mapCtx, in.Source.BigQueryDatasetSource)
+	return out
+}
+
+func Listing_BigQueryDatasetSource_ToProto(mapCtx *direct.MapContext, in *krmv1alpha1.BigQueryDatasetSource) *pb.Listing_BigqueryDataset {
+	if in == nil {
+		return nil
+	}
+
+	out := &pb.Listing_BigqueryDataset{}
+	if in.Dataset != nil {
+		out.BigqueryDataset = &pb.Listing_BigQueryDatasetSource{
+			Dataset: in.Dataset.External,
+		}
+
+		out.BigqueryDataset.RestrictedExportPolicy = &pb.Listing_BigQueryDatasetSource_RestrictedExportPolicy{}
+		if in.RestrictedExportPolicy != nil {
```

Comment:
What happens if we send `BigQueryDatasetSource` with empty Dataset but the other 2 fields are configured?

Reply:
I'd say let's let the underlying cloud api let error out rather than try to pre-optimize here

Reply:
The current code only send the other 2 fields if the field `Dataset` is configured, which is why I asked the first question. I agree we can delegate to the underlying API.